VirtoCommerce.Storefront/Controllers/Api/ApiCartController.cs

```diff
@@ -224,25 +224,25 @@ public async Task<ActionResult> AddCartCoupon(string couponCode)
             {
                 var cartBuilder = await LoadOrCreateCartAsync();
 
-                await cartBuilder.AddCouponAsync(couponCode);
+                var isUnique = await cartBuilder.AddCouponAsync(couponCode);
```

Comment:
Could you clarify why do we need this kind of check on this level?

Reply:
The idea was not to rely on front-end logic and to avoid returning wrong object when a duplicate coupon is filtered out by the "AddCouponAsync" check.
Currently the return value is not being used at all in default theme ("GetCart" is called), should we always return "Json(null)" instead?